### PR TITLE
S2166: Add C# and VB.NET

### PR DESCRIPTION
**Pull request by zsolt-kolbay-sonarsource**
https://github.com/SonarSource/sonar-dotnet/issues/6705

### CODE REVIEW

**Review by antonioaversa — COMMENTED** [review]
LGTM
@andrei-epure-sonarsource : your turn

File: rules/S2166/csharp/rule.adoc [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
+{
+  private Fruit expected;
+  private string unusualCharacteristics;
+  private bool appropriateForCommercialExploitation;
+  // ...
+}
+
+public class CarException: Exception // Compliant - correctly extends System.Exception
+{
+  public CarException(string message, Exception innerException): base(message, innerException)
```
Comment (andrei-epure-sonarsource) [inline rules/S2166/csharp/rule.adoc:38]: nitpick (same as [docs](https://learn.microsoft.com/en-us/dotnet/standard/exceptions/how-to-create-user-defined-exceptions))

same for Noncompliant and for VB .NET samples (:why: - less keystrokes, easier to read)

```suggestion
  public CarException(string message, Exception inner): base(message, inner)    
```

File: rules/S2166/csharp/rule.adoc [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+{
+  private Fruit expected;
+  private string unusualCharacteristics;
+  private bool appropriateForCommercialExploitation;
+  // ...
+}
+
+public class CarException // Noncompliant - does not derive from any Exception-based class
+{
+  public CarException(string message, Exception innerException)
```
Comment (andrei-epure-sonarsource) [inline rules/S2166/csharp/rule.adoc:17]:
```suggestion
  public CarException(string message, Exception inner) 
```

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM, with some minor nitpick suggestions.

**Pull request merged**